src/hypertable_restrict_info.c

```diff
@@ -595,11 +624,64 @@ gather_restriction_dimension_vectors(const HypertableRestrictInfo *hri)
 
 Chunk **
 ts_hypertable_restrict_info_get_chunks(HypertableRestrictInfo *hri, Hypertable *ht,
-									   LOCKMODE lockmode, unsigned int *num_chunks)
+									   unsigned int *num_chunks)
 {
-	List *dimension_vecs = gather_restriction_dimension_vectors(hri);
-	Assert(hri->num_dimensions == ht->space->num_dimensions);
-	return ts_chunk_scan_by_constraints(ht->space, dimension_vecs, lockmode, num_chunks);
+	/*
+	 * Filter out the dimensions for which we don't have a nontrivial
+	 * restriction.
```

Comment:
Btw, do we need to include "trivial" restrictions in `HypertableRestrictInfo` to begin with? Why not exclude them from the beginning so that we need not filter them out later?

Reply:
> Btw, do we need to include "trivial" restrictions in HypertableRestrictInfo to begin with? Why not exclude them from the beginning so that we need not filter them out later?

Originally I just didn't want to change the algorithm too much, because it would require different code for initializing the dimension restrict infos on demand.

But now I think this approach also has value for queries when there are multiple conditions on a dimension, that end up adding to the entire dimension. This isn't likely to arise in simple hand-written queries, but more likely in auto-generated queries or more complex queries that use views. So we can consider this approach of "removing empty ones in the end" to be an additional optimization.